### ISSUE TITLE
Attempting to manage binary files with Terraform github_repository_file provides erroneous results

### ISSUE THREAD

**Issue by mykelalvis** [integrations/terraform-provider-github#710]
We have a process that heavily utilizes Terraform to manage large swathes of infrastructure.

I am trying to use TF to setup GitHub repositories with files supplied as part of the standup process.

Text data is manageable, but binary data definitely is not.

### Terraform Version
```
Terraform v0.14.7
+ provider registry.terraform.io/hashicorp/github v4.5.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
```

### Affected Resource(s)
- `github_repository_file`

### Terraform Configuration Files
```hcl
resource "github_repository_file" "rando" {
  repository          = github_repository.randorepo.name
  branch              = "master"
  file                    =  "rando.pdf"
  content             = filebase64("../resources/rando.pdf")
  commit_message      = "Managed by Terraform"
  overwrite_on_create = true
}
```

### Debug Output
I am disallowed from revealing the information that a trace log provides.  However, I have attached the file to this issue and the results should be at least very similar.

### Panic Output
none

### Expected Behavior

I'd expect the base64 contents of the files specified to either
1. Be placed in the specified file on the repo instead of delivered as plain text 
OR
1. The provider to tell me that this is impossible

### Actual Behavior

For small files, the Base64 values arrived into the file as a stream of character.  This is an error.
For larger files there is an actual error message of failure:

```
2021-02-25T12:30:32.481-0600 [DEBUG] plugin.terraform-provider-github_v4.5.0: 2021/02/25 12:30:32 [DEBUG] Found file: rando1.zip in commit: a199949d9e6cc16537aa881f8b3248ed389a39f4
2021-02-25T12:30:32.481-0600 [DEBUG] plugin.terraform-provider-github_v4.5.0: 2021/02/25 12:30:32 [DEBUG] Found file: testing-community/rando/rando1.zip, in commit SHA: a199949d9e6cc16537aa881f8b3248ed389a39f4 
[31m
[1m[31mError: [0m[0m[1mProvider produced inconsistent result after apply[0m

[0mWhen applying changes to github_repository_file.randopdfile1, provider
"registry.terraform.io/hashicorp/github" produced an unexpected new value:
Root resource was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
[0m[0m
2021-02-25T12:30:32.489-0600 [DEBUG] plugin: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/github/4.5.0/linux_amd64/terraform-provider-github_v4.5.0 pid=39059
2021-02-25T12:30:32.489-0600 [DEBUG] plugin: plugin exited
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Nothing is remotely interesting about the target accounts

**Comment by mykelalvis** [comment]
[trial.zip](https://github.com/integrations/terraform-provider-github/files/6045127/trial.zip)

**Comment by Dilfer** [comment]
I am running into this same issue.

We use Gradle projects at our organization and I am trying to add the gradle/gradle-wrapper.jar file as part of our repo creation via terraform. Otherwise I am creating the repo, and then need to do a manual commit of the 4 relevant Gradle files after they are bootstrapped/downloaded by my IDE. 

I would love to be able to add binary files to my repos to get rid of this manual step.

**Comment by Dilfer** [comment]
@mykelalvis I am not sure if this workaround works for you folks, but it solved our intended use-case so I figured I would share. I believe this will only work if you want to get the files there initially but not necessarily manage their contents going forward.

I wanted a JAR file to be present at runtime but I don't need to update it going forward. So we ended up creating a template repository for all binary data we needed, and then we configured the template in the github_repository resource. All of the text files we need to manage the content of going forward, are all still maintained via the github_file resource like before and layered in after the repo is cloned from the template.
